api/base/serializers.py

```diff
@@ -1003,7 +1009,19 @@ def is_valid(self, clean_html=True, **kwargs):
         return ret
 
 
-class JSONAPISerializer(ser.Serializer):
+class PrefetchRelationshipsSerializer(ser.Serializer):
+    relationship_field_names = list()
+
+    def __init__(self, *args, **kwargs):
+        super(PrefetchRelationshipsSerializer, self).__init__(*args, **kwargs)
+        for field_name, field in self.fields.iteritems():
+            if field.source == '*':
+                self.relationship_field_names.append(field_name)
```

Comment:
Also, is it necessary to store all the field names on the serializer?

Wherever we would want to fetch an object with prefetching, couldn't we do something like

```python
prefetch_fields=Node.get_fk_field_names() + Node.get_m2m_field_names()
```

Of course, you could refactor that using a cached property, e.g. `Node.relationship_field_names`.